data/mods/aftershock_exoplanet/EOC/world_eocs.json

```diff
@@ -0,0 +1,33 @@
+[
+  {
+    "type": "effect_on_condition",
+    "id": "afs_spot_spaceport",
+    "recurrence": [ "8 hours", "48 hours" ],
+    "condition": {
+      "and": [
+        { "one_in_chance": 2 },
+        { "not": { "math": [ "has_var(know_spaceport)" ] } },
+        { "not": { "u_has_mission": "MISSION_REACH_SPACEPORT" } },
+        { "not": "is_day" },
+        "u_can_see",
+        { "is_weather": "clear" },
+        "u_is_outside"
+      ]
+    },
+    "effect": [
+      { "assign_mission": "MISSION_REACH_SPACEPORT" },
+      {
+        "u_message": "You notice the rocket plumes of three ascending ships in the distance.  There must be a large space port in that direction."
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_YOU_KNOW_SPACEPORT",
+    "global": true,
+    "eoc_type": "EVENT",
+    "required_event": "avatar_enters_omt",
+    "condition": { "and": [ { "not": { "math": [ "has_var(know_spaceport)" ] } }, { "u_at_om_location": "land_pad_outpost_a1" } ] },
```

Comment:
If you just want to set variable, as i said in discord, use entry_eoc
https://github.com/CleverRaven/Cataclysm-DDA/blob/ace355a82b185d963369ceb8165967ab2042df05/data/mods/Magiclysm/worldgen/overmap_terrain.json#L678
if you want to give some room, use `u_near_om_location` instead of `u_at_om_location`

Reply:
Ah nice.